addons/mail/static/src/models/rtc/rtc.js

```diff
@@ -0,0 +1,254 @@
+         * @param {Boolean} [param0.audio]
+         * @returns {Stream} output from the media devices.
+         */
+        async _getStream({ video=true, audio=true } = {}) {
+            try {
+                return await navigator.mediaDevices.getUserMedia({
+                    video,
+                    audio,
+                });
+            } catch (e) {
+                // fallback on Audio-only (happens if the device doesn't have a camera or doesn't allow its utilization.
+                return await navigator.mediaDevices.getUserMedia({
+                    audio,
+                });
+            }
+        }
+
+        /**
+         *
+         * @private
+         * @param {String} state the new state of the connection
+         * @param {String} token of the peer whose the connection changed
+         */
+        async _onConnectionStateChange(state, token) {
+            switch(state) {
+                case "failed":
+                case "closed":
+                case "disconnected":
+                    this._removePeer(token);
+            }
+        }
+
+        /**
+         * process the media connection of new calls
+         *
+         * @private
+         * @param {Object} mediaConnection
+         */
+         _processCall(mediaConnection) {
+            mediaConnection.answer(this.stream);
+            mediaConnection.on('stream', callerStream => {
+                const callToken = mediaConnection.peer;
+                this._addStream(callerStream, callToken);
+                mediaConnection.peerConnection.onconnectionstatechange = () => {
+                    this._onConnectionStateChange(mediaConnection.peerConnection.connectionState, callToken);
+                };
+            });
+            mediaConnection.on('error', error => {
+                console.log(error);
+            });
+        }
+
+        /**
+         * cleans up a peer and its video
+         *
+         * @private
+         * @param {String} token
+         */
+        _removePeer(token) {
+            //this.room.removeUser(token);
+            const newActivePeers = Object.assign({}, this.activePeers);
+            delete newActivePeers[token];
+            this.update({ activePeers: newActivePeers });
+        }
+
+        /**
+         * @private
+         * @param {Object} peer the peerObject of the current partner.
+         */
+        _setupPeer(peer) {
+            peer.on('call', mediaConnection => {
+                this._processCall(mediaConnection);
+            });
+            peer.on('error', error => {
+                console.log('PEER-ERROR:::::');
+                console.log(error);
+            });
+        }
+    }
+
+    Rtc.fields = {
+        peer: attr(),
+        /*
+         * Object that contains the peer streams per peer token
+         * { token: { token, stream }}
+         *
+         */
+        activePeers: attr(),
+        peerToken: attr({
+            default: '',
+        }),
```

Comment:
I have some issue understand what is a "peer". Maybe you need `UUID` on partner and rely on relations instead? The data structure of `activePeers` looks complicated, which IMO clearly justifies making use of relations instead of just this attr field.